frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -271,6 +271,14 @@ const ChildrenBody = ({
       ) {
         return ''
       }
+      // We use the childname to indicate if there is a child below 21.
+      // If the childname is an empty string, it means there is no child below 21.
+      // As our definition of child in FormSG means child below 21, we want to
+      // return empty strings for other child attributes even if their value is populated by myinfo
+      // if there is no childname.
+      if (myInfoChildrenBirthRecords.childname?.[0] === '') {
+        return ''
```

Comment:
Can we move this check above to before `result` is computed?

Reply:
yes! done in [e35bc74](https://github.com/opengovsg/FormSG/pull/6715/commits/e35bc74e517f0eb7a00dc697ceda07bd3b5e35a1)